### PR TITLE
NanoUI Shit (Part 3) - JavaShit

### PR DESCRIPTION
**Pull request by tigercat2000**
- Nano's JavaScript has been cleaned up significantly (mostly spacing changes)
 - Made NanoUI errors easier to debug, VV a client's topic_debugging variable to 1 for them to receive Nano errors.
 - Moved concat order into a new plugin, `gulp-order`. Renamed all JavaScript files back to their original names. Instead of concatenating based strictly on file name, it will now also follow the order set out in gulpfile.  Currently, all this order does is assure the following:
  - JQuery is loaded before JQuery's plugins, JQuery's plugins are loaded before doT.js
  - NanoUtility and NanoStateManager are loaded before any other Nano JS file, as they both define important methods and functions, which are not able to be hoisted as normal. This includes the `inheritsFrom` property and NanoStateManager being available for registering NanoStates(s)

### CODE REVIEW

File: nano/scripts/nano/nano_template.js [hunk 1]
```diff
@@ -5,19 +5,19 @@ var NanoTemplate = function () {
...
-		_templateData = $('body').data('templateData');
+        _templateData = $('body').data('templateData');
 
-		if (_templateData == null)
-		{
-			alert('Error: Template data did not load correctly.');
-		}
+        if (_templateData == null)
+        {
+            reportError('Error: Template data did not load correctly.');
```
Comment (tigercat2000) [inline nano/scripts/nano/nano_template.js]: I can't guarantee if this function works at all, as I do not have any consistent way to produce NanoUI errors.

Comment (Tastyfish) [inline nano/scripts/nano/nano_template.js]: Writing something invalid like `{{:(}hello` should spam constant nanoui errors in a loop.

Comment (tigercat2000) [inline nano/scripts/nano/nano_template.js]: if I had any idea how doT errors worked, I would love to do that

Comment (Tastyfish) [inline nano/scripts/nano/nano_template.js]: Aw, I was hoping it would cover DOT errors too, since those are so easy to make.

Comment (tigercat2000) [inline nano/scripts/nano/nano_template.js]: i have no idea if doT even has error handling for templates

Comment (Tastyfish) [inline nano/scripts/nano/nano_template.js]: Yeah I don't think it does, it just awkwardly clobbers the output and then the runtime state manager gives the generic error, iirc.

**Comment by tigercat2000** [comment]
i missed the opportunity to name this "ShitScript"
i am disappointed

**Comment by Spacemanspark** [comment]
I guess you could say the cat's out of the bag.
I'll go settle this over a cup of Java.

**Pull request merged**